### DISCUSSION TITLE
License change

### DISCUSSION THREAD

**Discussion by jamaa** [bluemodel/BlueM.Wave#78] [Polls]
It's been a long time in the making but BlueM.Wave is finally (really) going open-source. Time to decide on the license.

Until now, BlueM.Wave has been published (in binary form only) under the quite permissive [BSD-2-Clause license](https://opensource.org/licenses/BSD-2-Clause). Once the code is open-source, we want to ensure that any modifications of the code stay open-source, which is why we need to change the license to one which enforces this.

The options we have decided on are :
* [GNU General Public License (GPLv3)](https://www.gnu.org/licenses/gpl-3.0.en.html)
* [GNU Lesser General Public License (LGPLv3)](https://www.gnu.org/licenses/lgpl-3.0.html)

The main difference between the two being that the LGPLv3 would allow using Wave as a library in any other software, while the GPLv3 would mean that Wave can only be used as a library in other software if that software is also open-source.

Or, as phrased by https://choosealicense.com/licenses/lgpl-3.0/ :
> Permissions of this copyleft license are conditioned on making available complete source code of licensed works and modifications under the same license or the GNU GPLv3. [...] However, a larger work using the licensed work through interfaces provided by the licensed work may be distributed under different terms and without source code for the larger work. 

@bluemodel/bluem-devs you may cast your vote below.

**Comment by jamaa** [comment]
I vote for LGPLv3, because I would like BlueM.Wave to be usable as a library even by proprietary software such as e.g. [Talsim-NG](http://www.talsim.de). I believe the stricter GPLv3 would restrict possible adoption of Wave as a supplementary tool, which is already a quite small niche anyway. 

The last few years of development in Wave were primarily driven by SYDRO's desire to use Wave as a supplementary tool to Talsim. The LGPLv3 would open more opportunities for the use of Wave as a library within Talsim and further drive the future development of Wave.

**Comment by jamaa** [comment]
Either nobody cares, or the poll system doesn't work :-)

I guess LGPLv3 it is then.

**Reply by wassermax** [reply]
Sorry for my delayed poll reply! I have not voted – and you could probably guess my vote :-) However, I fully see the point of you and SYDRO's desire to make the best use of the BlueM.Wave as library in Talsim. Maybe we could make BlueM.Wave release publicly under the GPLv3 license and grant interested close BlueM cooperation partners (e.g., SYDRO) the LGPLv3 license.

**Reply by wassermax** [reply]
Anyhow, if this is too complicated, I can also live with the LGPLv3 license for BlueM.Wave. my main concern is, that it should be clearly visible what part of the bundled software is BlueM.Wave (and thus based on long years of work of the Dev Team) to gain more interest and attraction for the whole BlueM ecosystem.

**Reply by jamaa** [reply]
Thanks for your input. I can assure you that Wave, even if bundled with Talsim, will still be clearly recognizable as BlueM.Wave.

**Reply by jamaa** [reply]
To clarify, "bundling" Wave means using the [Wave API](https://wiki.bluemodel.org/index.php/Wave:API) e.g. for reading and displaying timeseries. When used this way, Wave always opens in its own new window. Just like how BlueM.Win and BlueM.Opt use Wave. As far as I know, it is not possible to embed Wave or parts of it in another GUI. It will always be shown as a separate window. So the risk of mistaking it for an integral part of the other software is low.

**Comment by jamaa** [comment]
We now have two votes, one for each option, and @wassermax said he did not vote but "can also live" with the LGPL. So can we end the discussion and decide on the LGPL?

**Comment by devmki** [comment]
Hi Felix
Sorry for the very late reply.
LGPLv3 is fine with me.

Perhaps as an idea: Does wave have an "about" button to direct to the bluem
project?

Cheers
Michael

Felix Froehlich ***@***.***> schrieb am Fr., 4. Nov. 2022,
17:01:

**Reply by jamaa** [reply]
Thanks for the feedback. Yes Wave does have an about box, and it has BlueM all over it
![grafik](https://user-images.githubusercontent.com/90166/200033690-448724a7-7057-4ad4-8e1d-a0f306938520.png)

**Comment by devmki** [comment]
That should do it :)


Felix Froehlich ***@***.***> schrieb am Fr., 4. Nov. 2022,
18:04:

**Comment by jamaa** [comment]
So. it seems that everyone is OK with LGPLv3. Great! I will make the necessary changes to the source files and then make the repo public. I will also bump the version number to v2.0.

**Comment by SHeusch** [comment]
I’m fine with it, Steffen.

Sorry as well for not replying earlier…

 

-----

Prof. Dr.-Ing. Steffen Heusch

TH Mittelhessen

FG Wasserwirtschaft und Hydrologie

ZEuUS - Kompetenzzentrum für nachhaltiges Engineering und UmweltSysteme

Wiesenstr. 14

35390 Gießen

Telefon: +(49) 641 309-1862

e-mail:  ***@***.***> ***@***.***

Zoom:  <https://thm-de.zoom.us/my/heusch> https://thm-de.zoom.us/my/heusch; Meeting-ID: 926 475 2113

 

Von: Felix Froehlich ***@***.***> 
Gesendet: Samstag, 5. November 2022 15:58
An: bluemodel/BlueM.Wave ***@***.***>
Cc: SHeusch ***@***.***>; Team mention ***@***.***>
Betreff: Re: [bluemodel/BlueM.Wave] License change (Discussion #78)

 

So. it seems that everyone is OK with LGPLv3. Great! I will make the necessary changes to the source files and then make the repo public. I will also bump the version number to v2.0.

**Comment by jamaa** [comment]
It's done!
